system/lib/webgpu/webgpu.cpp

```diff
@@ -81,3 +81,12 @@ void wgpuSurfaceGetCapabilities(WGPUSurface surface,
 void wgpuSurfaceCapabilitiesFreeMembers(WGPUSurfaceCapabilities value) {
   // wgpuSurfaceCapabilities doesn't currently allocate anything.
 }
+
+// WGPUAdapterInfo
+
+void wgpuAdapterInfoFreeMembers(WGPUAdapterInfo value) {
+  delete[] value.vendor;
+  delete[] value.architecture;
+  delete[] value.device;
+  delete[] value.description;
```

Comment:
In C++, `malloc` must be used with `free` and `new` must be used with `delete`. So to match the use of `_malloc` from JS, these must use `free` instead of `delete`.

Reply:
I'm not sure how to call `free` in cpp with `value.vendor` and so on.

Reply:
Should just be `free(value.vendor);`
where `free` comes from `#include <cstdlib>`

Reply:
That's what I tried but this resulted in the following error:

```
../../../system/lib/webgpu/webgpu.cpp:89:3: error: no matching function for call to 'free'
   89 |   free(value.vendor);
      |   ^~~~
/Users/fbeaufort/github/emscripten/cache/sysroot/include/sched.h:85:6: note: candidate function not viable: no known conversion from 'const char *' to 'void *' for 1st argument
   85 | void free(void *);
```

Reply:
I think its safe to `const_cast` that away.

Reply:
Ah... yes, you'll need to use `const_cast` to make the pointer non-const.

Reply:
Thanks!